crates/ruff_python_ast/src/node.rs

```diff
@@ -2582,16 +2589,53 @@ impl AstNode for ast::ExprFormattedValue {
     where
         V: PreorderVisitor<'a> + ?Sized,
     {
-        let ast::ExprFormattedValue {
-            value, format_spec, ..
+        let ast::FormattedValue {
+            expression,
+            format_spec,
+            ..
         } = self;
-        visitor.visit_expr(value);
+        visitor.visit_expr(expression);
 
-        if let Some(expr) = format_spec {
-            visitor.visit_format_spec(expr);
+        for spec_part in format_spec {
+            visitor.visit_fstring_part(spec_part);
         }
     }
 }
+impl AstNode for ast::PartialString {
+    fn cast(kind: AnyNode) -> Option<Self>
+    where
+        Self: Sized,
+    {
+        if let AnyNode::PartialString(node) = kind {
+            Some(node)
+        } else {
+            None
+        }
+    }
+
+    fn cast_ref(kind: AnyNodeRef) -> Option<&Self> {
+        if let AnyNodeRef::PartialString(node) = kind {
+            Some(node)
+        } else {
+            None
+        }
+    }
+
+    fn as_any_node_ref(&self) -> AnyNodeRef {
+        AnyNodeRef::from(self)
+    }
+
+    fn into_any_node(self) -> AnyNode {
+        AnyNode::from(self)
+    }
+
+    fn visit_preorder<'a, V>(&'a self, _visitor: &mut V)
+    where
+        V: PreorderVisitor<'a> + ?Sized,
+    {
+        // TODO: is this correct?
+    }
```

Comment:
Yes, I think it's correct. (You can inline this method)
```suggestion
	#[inline]
    fn visit_preorder<'a, V>(&'a self, _visitor: &mut V)
    where
        V: PreorderVisitor<'a> + ?Sized,
    {
    }
```

Reply:
interestingly for `ExprConstant` we have a `visit_constant` that is empty instead